src/main/java/com/simibubi/create/api/contraption/storage/item/MountedItemStorageWrapper.java

```diff
@@ -10,12 +10,48 @@
 /**
  * Wrapper around many MountedItemStorages, providing access to all of them as one storage.
  * They can still be accessed individually through the map.
+ * 
+ * Uses O(1) lookup arrays instead of O(n) linear scan.
```

Comment:
I have a feeling this optimization is doing the bulk of the heavy lifting in the results you’ve demonstrated. You don’t have to, but I’d be curious to see the diff with just the slot LUT